listenbrainz/mbid_mapping_writer/mbid_mapper.py

```diff
@@ -0,0 +1,284 @@
+
+            query = " ".join(cleaned_query)
+
+        search_parameters = {
+            'q': query,
+            'query_by': "combined",
+            'prefix': 'no',
+            'num_typos': self.MATCH_TYPE_MED_QUALITY_MAX_EDIT_DISTANCE
+        }
+
+        while True:
+            try:
+                hits = self.client.collections[COLLECTION_NAME].documents.search(
+                    search_parameters)
+                break
+            except requests.exceptions.ReadTimeout:
+                print("Got socket timeout, sleeping 5 seconds, trying again.")
+                sleep(5)
+
+        if len(hits["hits"]) == 0:
+            return None
+
+        return hits["hits"][0]
+
+    def search(self, artist_credit_name, recording_name):
+        """
+            Main query body: Prepare the search query terms and prepare
+            detuned query terms. Then attempt to find the given search terms
+            and if not found, sequentially try the detuned versions of the
+            query terms. Return a match dict (properly formatted for this
+            query) or None if not match.
+        """
+
+        artist_credit_name_p = prepare_query(artist_credit_name)
+        recording_name_p = prepare_query(recording_name)
+
+        ac_detuned = prepare_query(self.detune_query_string(artist_credit_name, True))
+        r_detuned = prepare_query(self.detune_query_string(recording_name, False))
+        self._log(f"ac_detuned: '{ac_detuned}' r_detuned: '{r_detuned}'")
+
+        self.is_ac_detuned = False
+        self.is_r_detuned = False
+
+        while True:
+            hit = self.lookup(artist_credit_name_p, recording_name_p)
+            if hit:
+                (hit, match_type) = self.evaluate_hit(
+                    hit, artist_credit_name_p, recording_name_p)
+                if hit:
+                    self._log(Markup(f"""
+                        <table>
+                            <tr>
+                                <th>recording name</th>
+                                <th>release name</th>
+                                <th>artist credit name</th>
+                                <th>recording MBID</th>
+                                <th>release MBID</th>
+                                <th>artist_credit_id</th>
+                            </tr>
+                            <tr>
+                                <td>{hit['document']["recording_name"]}</td>
+                                <td>{hit['document']["release_name"]}</td>
+                                <td>{hit['document']["artist_credit_name"]}</td>
+                                <td>{hit['document']["recording_mbid"]}</td>
+                                <td>{hit['document']["release_mbid"]}</td>
+                                <td>{hit['document']["artist_credit_id"]}</td>
+                            </tr>
+                        </table>
+                    """))
+
+            if not hit:
```

Comment:
My understanding is that this covers three cases:
1) original names
2) ac detuned r original
3) ac detuned r detuned

Do you think it makes sense to add the 4th, ac original r detuned?